python/python-psi-impl/src/com/jetbrains/python/psi/resolve/PyResolveUtil.java

```diff
@@ -360,7 +360,7 @@ public static PyExpression fullResolveLocally(@NotNull PyReferenceExpression ref
    */
   public static boolean allowForwardReferences(@NotNull PyQualifiedExpression element) {
     // Allow forward references in Pyi annotations
-    if (PyiUtil.isInsideStubAnnotation(element)) {
+    if (PyiUtil.isInsideStubAnnotation(element) || PyiUtil.isInsideStub(element)) {
```

Comment:
your check covers the first part of expression, let's simplify it (and update comment)

please add a test to com.jetbrains.python.pyi.PyiInspectionsTest or com.jetbrains.python.pyi.PyiResolveTest

Reply:
Are you suggesting to simply remove the first check?

Reply:
yes, since `isInsideStubAnnotation` implies `isInsideStub`